block/s3/inventory_iterator.go

```diff
@@ -92,13 +96,17 @@ func (it *InventoryIterator) moveToNextManifestFile() bool {
 }
 
 func (it *InventoryIterator) fillBuffer() bool {
-	pr, closeReader, err := it.getParquetReader(it.ctx, it.S3, it.inventoryBucket, it.Manifest.Files[it.currentManifestFileIdx].Key)
+	key := it.Manifest.Files[it.currentManifestFileIdx].Key
+	pr, closeReader, err := it.getParquetReader(it.ctx, it.S3, it.inventoryBucket, key)
 	if err != nil {
 		it.err = err
 		return false
 	}
 	defer func() {
-		_ = closeReader()
+		err = closeReader()
+		if err != nil {
+			it.logger.Errorf("failed to close parquet reader in fillBuffer. bucket=%s, key=%s", it.inventoryBucket, key)
```

Comment:
AFAICT this still swallows errors on close: they are reported but ultimately ignored (and the import "succeeds").  Can we set a failure bit here?  Alternatively the usual approach is _not_ to close handles during a defer, and then return errors properly.

Assuming Parquet readers don't add new close-fail behaviours, this should happen only if the filesystem close fails.  Parquet `ReadStop` is a bit cavalier about read failures, so they probably don't believe in close failure.  Unfortunately the Linux `man 2 close` page doesn't give me any realistic failure cases (for readonly file descriptors).  It gives:
* `EBADF`, which is a programming error
* `EINTR`, which is due to signal handling
* `EIO`, an I/O error, but since we're closing we believe we have all the data
* `ENOSPCS`, `EDQUOT`, which cannot happen for readonly files.

The most realistic scenario I can imagine is importing from  NFS or SMB or some other network-mounted filesystem.  That is unlikely _but possible_.

Reply:
Thing is, I don't really want to fail the import in this case, because here the file has already been read in its entirety, so the import is on track.

Reply:
@arielshaqed @johnnyaug in this case, reading is scoped to parquet files on S3 only.
Not sure what closing even really means in this context.